pcap4j-core/src/main/java/org/pcap4j/packet/GtpV1PduSessionContainerExtensionHeader.java

```diff
@@ -0,0 +1,490 @@
+            GtpV1ExtensionHeaderType.getInstance(
+                (byte) (paddingAndNextExtensionHeaderType & 0x000000FF));
+      } else {
+        this.nextExtensionHeaderType =
+            GtpV1ExtensionHeaderType.getInstance(
+                ByteArrays.getByte(rawData, NEXT_EXTENSION_HEADER_TYPE_OFFSET + offset));
+      }
+    } else if (pduType == 1) {
+      byte spareAndQfi = ByteArrays.getByte(rawData, SPARE_AND_QFI_OFFSET + offset);
+      this.qfi = (byte) (spareAndQfi & 0x3F);
+      this.nextExtensionHeaderType =
+          GtpV1ExtensionHeaderType.getInstance(
+              ByteArrays.getByte(rawData, NEXT_EXTENSION_HEADER_TYPE_OFFSET + offset));
+    }
+  }
+
+  private GtpV1PduSessionContainerExtensionHeader(Builder builder) {
+    this.pduType = builder.pduType;
+    this.ppp = builder.ppp;
+    this.rqi = builder.rqi;
+    this.qfi = builder.qfi;
+    this.ppi = builder.ppi;
+    this.padding = builder.padding;
+    this.nextExtensionHeaderType = builder.nextExtensionHeaderType;
+
+    if (builder.correctLengthAtBuild) {
+      this.length = 1;
+      if (pduType == 0 && ppp) {
+        this.length = 2;
+      }
+    } else {
+      this.length = builder.length;
+    }
+  }
+
+  /** @return length */
+  public byte getLength() {
+    return length;
+  }
+
+  /** @return length as int */
+  public int getLengthAsInt() {
+    return 0xFF & length;
+  }
+
+  /** @return pduType */
+  public byte getPduType() {
+    return pduType;
+  }
+
+  /** @return ppp */
+  public boolean isPpp() {
+    return ppp;
+  }
+
+  /** @return rqi */
+  public boolean isRqi() {
+    return rqi;
+  }
+
+  /** @return qfi */
+  public byte getQfi() {
+    return qfi;
+  }
+
+  /** @return ppi */
+  public byte getPpi() {
+    return ppi;
+  }
+
+  /** @return padding */
+  public int getPadding() {
+    return padding;
+  }
+
+  /** @return nextExtensionHeaderType */
+  @Override
+  public GtpV1ExtensionHeaderType getNextExtensionHeaderType() {
+    return nextExtensionHeaderType;
+  }
+
+  @Override
+  public byte[] getRawData() {
+    List<byte[]> rawFields = new ArrayList<byte[]>();
+    rawFields.add(ByteArrays.toByteArray(length));
+    rawFields.add(ByteArrays.toByteArray((byte) (pduType << 4)));
+    if (pduType == 0) {
+      byte firstByte = 0;
+      int lastInt = 0;
+      if (ppp) {
+        firstByte |= 0x80;
+        lastInt = (ppi << 28) | (padding & 0x00FFFFFF);
+      }
+      if (rqi) {
+        firstByte |= 0x40;
+      }
+      firstByte |= qfi & 0x3F;
+      rawFields.add(ByteArrays.toByteArray(firstByte));
+      rawFields.add(ByteArrays.toByteArray(lastInt));
+    } else if (pduType == 1) {
+      byte oneByte = (byte) (qfi & 0x3F);
+      rawFields.add(ByteArrays.toByteArray(oneByte));
+    }
+    rawFields.add(ByteArrays.toByteArray(nextExtensionHeaderType.value() & 0xFF));
+    return ByteArrays.concatenate(rawFields);
+  }
+
+  @Override
+  public int length() {
+    return getLengthAsInt() * 4;

```

Comment:
`length()` must return an actual data length.
The value of length field can be set by users and we can't trust it.